lib/pbench/agent/tool_meister.py

```diff
@@ -73,6 +112,245 @@ class ToolException(Exception):
+        self.logger = logger
+        self.tool_pmda = tool_pmda
+        # initialize internal variables
+        self.roles_path = None
+        # first find the roles path
+        self.find_roles_path()
+        # second find if performance co-pilot role has been installed or not
+        # install if not
+        self.downloaded_role = False
+        self.download_role()
+
+        # ansible part
+        context.CLIARGS = ImmutableDict(connection='local')
+        # initialize needed objects
+        self.loader = (
+            DataLoader()
+        )  # Takes care of finding and reading yaml, json and ini files
+        self.passwords = dict(vault_pass="secret")
+
+        # create inventory, use path to host config file as source or hosts in a comma separated string
+        self.inventory = InventoryManager(loader=self.loader, sources=sources='localhost,')
+
+        # variable manager takes care of merging all the different sources to give you a unified view of variables available in each context
+        self.variable_manager = VariableManager(
+            loader=self.loader, inventory=self.inventory
+        )
+
+    def find_roles_path(self):
+        """ find where roles are stored """
+        temp = None
+        resp = None
+
+        try:
+            with open(ANSIBLE_CONF, mode="r") as conf_file:
+                temp = conf_file.read()
+                self.logger.debug("contents of ansible conf file:%s", temp)
+        except FileNotFoundError:
+            self.logger.exception("ansible conf file does not exist")
+
+        for line in temp.split("\n"):
+            if "roles_path" in line:
+                resp = line
+
+        if resp is not None:
+            self.roles_path = resp.split("=")[-1].strip()
+            self.logger.debug("roles path:%s", self.roles_path)
+        else:
+            self.roles_path = ANSIBLE_ROLES_PATH
+            self.logger.error("not able to find roles path. setting it to default")
+
+    def download_role(self):
+        """ Download role if it does not exist in the localhost """
+        # build command for finding contents of roles path folder
+        command = "ls -lA " + self.roles_path
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        # decode the output
+        output = stdout.decode()
+        self.logger.debug(
+            "output from command:%s is:%s\n and stderr:%s",
+            command,
+            stdout.decode(),
+            stderr,
+        )
+
+        # find if the required role exists or not
+        for line in output:
+            if "performancecopilot.metrics.pcp" in line:
+                # if it exists, return. Dont have to proceed
+                return
+
+        # given role does not exist. We have to download it from ansible galaxy
+        command = "ansible-galaxy collection install performancecopilot.metrics"
```

Comment:
As we talked about today verbally, let's drop the use of Ansible to install and start PCP on local/remote systems.  Philosophically, the pbench-agent should not be in the business of install software on hosts, or changing the host configuration to suite its needs.  Instead let's build a container for `pmcd` which encapsulates what we need, and do the same for `node-exporter` as well.